test/divider_test.dart

```diff
@@ -0,0 +1,428 @@
+        final in2 = lastB[0];
+        lastA.removeAt(0);
+        lastB.removeAt(0);
+        if (triggerCheck) {
+          final check = (in2 == 0) ? divZero : ((in1 ~/ in2) == currResult);
+          if (check) {
+            String msg = (in2 == 0)
+                ? 'Divide by 0 error correctly encountered for denominator of 0.'
+                : 'Correct result: a=$in1, b=$in2, result=$currResult';
+            logger.info(msg);
+          } else {
+            String msg = (in2 == 0)
+                ? 'No Divide by zero error for denominator of 0.'
+                : 'Incorrect result: a=$in1, b=$in2, result=$currResult';
+            logger.severe(msg);
+          }
+          triggerCheck = false;
+        }
+      }
+    });
+  }
+}
+
+class DivAgent extends Agent {
+  final DivInterface intf;
+  late final DivSequencer sequencer;
+  late final DivDriver driver;
+  late final DivInputMonitor inMonitor;
+  late final DivOutputMonitor outMonitor;
+
+  DivAgent(this.intf, Component parent, {String name = 'divAgent'})
+      : super(name, parent) {
+    sequencer = DivSequencer(this);
+    driver = DivDriver(intf, sequencer, this);
+    inMonitor = DivInputMonitor(intf, this);
+    outMonitor = DivOutputMonitor(intf, this);
+  }
+}
+
+class DivEnv extends Env {
+  final DivInterface intf;
+
+  late final DivAgent agent;
+  late final DivScoreboard scoreboard;
+
+  DivEnv(this.intf, Component parent, {String name = 'divEnv'})
+      : super(name, parent) {
+    agent = DivAgent(intf, this);
+    scoreboard = DivScoreboard(
+        agent.inMonitor.stream, agent.outMonitor.stream, intf, this);
+  }
+
+  @override
+  Future<void> run(Phase phase) async {
+    unawaited(super.run(phase));
+  }
+}
+
+class DivBasicSequence extends Sequence {
+  DivBasicSequence({String name = 'divBasicSequence'}) : super(name);
+
+  @override
+  Future<void> body(Sequencer sequencer) async {
+    final divSequencer = sequencer as DivSequencer;
+
+    divSequencer.add(
+        DivInputSeqItem(mA: 4, mB: 2, mNewInputs: true)); // even divide by 2
+    divSequencer.add(DivInputSeqItem(
+        mA: 9, mB: 3, mNewInputs: true)); // even divide not by 2
+    divSequencer.add(
+        DivInputSeqItem(mA: 5, mB: 2, mNewInputs: true)); // not even divide
+    divSequencer
+        .add(DivInputSeqItem(mA: 4, mB: 1, mNewInputs: true)); // divide by 1
+    divSequencer.add(
+        DivInputSeqItem(mA: -10, mB: 2, mNewInputs: true)); // negative-positive
+    divSequencer.add(DivInputSeqItem(
+        mA: 13, mB: -10, mNewInputs: true)); // positive-negative
+    divSequencer.add(DivInputSeqItem(
+        mA: -10, mB: -9, mNewInputs: true)); // negative-negative
+    divSequencer
+        .add(DivInputSeqItem(mA: 1, mB: 4, mNewInputs: true)); // bigger divisor
+    divSequencer
+        .add(DivInputSeqItem(mA: 4, mB: 0, mNewInputs: true)); // divide by 0
+  }
+}
+
+class DivVolumeSequence extends Sequence {
+  final int numReps;
+  final rng = Random();
```

Comment:
let's use a fixed seed in tests

Reply:
I wonder if these test structures could be generalized for any component supplying a ready/valid on the upstream side.  I use very simple peek/poke testing with expect because my components are combinational, but I see a lot of code here that does more than division specific queueing of tests.  Does ROHD-VF provide something like this or can be generalized?

Reply:
Agreed - added a fixed seed